src/plugins/mer/mersigninguserselectiondialog.cpp

```diff
@@ -0,0 +1,73 @@
+#include "mersigninguserselectiondialog.h"
+#include <coreplugin/messagemanager.h>
+#include <sfdk/utils.h>
+#include <utils/algorithm.h>
+#include <QInputDialog>
+#include <QMessageBox>
+
+using namespace Sfdk;
+
+namespace {
+
+QList<GpgKeyInfo> getAvailableKeys()
+{
+    QList<GpgKeyInfo> availableKeys;
+    bool ok;
+    QString error;
+    execAsynchronous(std::tie(ok, availableKeys, error), Sfdk::availableGpgKeys);
+    if (!ok)
+        Core::MessageManager::write(error, Core::MessageManager::Flash);
+
+    return availableKeys;
+}
+
+} // namespace
+namespace Mer {
+namespace Internal {
+
+MerSigningUserSelectionDialog::MerSigningUserSelectionDialog(QObject *parent)
+    : QObject(parent)
+{}
+
+GpgKeyInfo MerSigningUserSelectionDialog::selectSigningUser()
+{
+    QList<GpgKeyInfo> availableKeys;
+    QString gpgErrorString;
+    if (Sfdk::isGpgAvailable(gpgErrorString)) {
```

Comment:
Swap condition, return, omit `else`, declare and initialize `const availableKeys` later in one step.